bug: PR Triage workflow fails - missing pr_triage.py script

## Bug Description
The PR Triage workflow (`PM - PR Triage`) is failing on all pull requests due to a missing automation script.

## Error Details
**Workflow**: PM - PR Triage  
**Run**: https://github.com/rysweet/MicrosoftHackathon2025-AgenticCoding/actions/runs/19618423712  
**Affected PR**: #1544 (and likely all PRs)

### Error Message
```
python: can't open file '/home/runner/work/MicrosoftHackathon2025-AgenticCoding/MicrosoftHackathon2025-AgenticCoding/.github/scripts/pr_triage.py': [Errno 2] No such file or directory
##[error]Process completed with exit code 2.
```

## Root Cause
The workflow file references `.github/scripts/pr_triage.py` but this script does not exist in the repository.

**Workflow Configuration** (`.github/workflows/pm-pr-triage.yml`):
```yaml
- name: Triage PR with Claude Agent SDK
  run: |
    python .github/scripts/pr_triage.py
  env:
    ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
    PR_NUMBER: ${{ github.event.pull_request.number }}
    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

## Impact
- **Severity**: Medium (workflow always fails but doesn't block PRs)
- **Critical CI checks still pass**: Code validation and security checks work correctly
- **User Experience**: Red X on PRs is confusing and annoying
- **Actual Risk**: Low - this is an automation enhancement, not a blocker

## Expected Behavior
The triage workflow should either:
1. Have the missing `pr_triage.py` script implemented
2. Be temporarily disabled until the script is ready
3. Fail gracefully with a clear message

## Reproduction
1. Create any PR
2. Observe CI checks
3. PR Triage workflow will fail with "No such file or directory"

## Environment
- **Workflow**: `.github/workflows/pm-pr-triage.yml`
- **Missing File**: `.github/scripts/pr_triage.py`
- **Python Version**: 3.11.14 (from workflow logs)
- **Required Environment Variables**: 
  - `ANTHROPIC_API_KEY` ✅ (present)
  - `PR_NUMBER` ✅ (present)
  - `GH_TOKEN` ✅ (present)

## Suggested Fix
Two options:

### Option 1: Implement the Script (Proper Fix)
Create `.github/scripts/pr_triage.py` that:
1. Uses Claude Agent SDK to analyze PR
2. Checks for:
   - Related issues
   - Test coverage
   - Breaking changes
   - Documentation updates
3. Posts triage summary as PR comment
4. Labels PR appropriately

### Option 2: Temporary Disable (Quick Fix)
Disable the workflow until script is implemented:
- Add `if: false` to workflow
- Or delete `.github/workflows/pm-pr-triage.yml`
- Re-enable after script is ready

## Related Context
- This workflow was likely added as part of M3 milestone automation
- Security notes in workflow suggest this is for prompt injection validation
- Claude Agent SDK integration is available and configured

## Acceptance Criteria
- [ ] Triage workflow either passes or is disabled
- [ ] No confusing red X on PRs from missing scripts
- [ ] If implemented, triage provides value (labels, comments, analysis)
- [ ] If disabled, clear documentation explains why

## Priority
**Medium** - Annoying but not blocking development

## Labels
- `bug`
- `ci/cd`
- `workflow`
- `M3-automation`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

--- a/.github/scripts/pr_triage/README.md
+++ b/.github/scripts/pr_triage/README.md
@@ -1,0 +1,228 @@
+# PR Triage Automation
+
+Automated PR validation, compliance checking, and triage reporting for GitHub pull requests.
+
+## Overview
+
+This package provides automated PR triage that validates workflow compliance, detects priority/complexity, identifies unrelated changes, and posts comprehensive triage reports as PR comments.
+
+## Architecture
+
+### MVP Implementation (Default)
+
+By default, the system uses **heuristic-based analysis** for fast, deterministic validation without requiring Claude CLI:
+
+```
+pr_triage/
+├── __init__.py              # Public interface
+├── validator.py             # Main orchestrator
+├── analyzers_mvp.py         # Heuristic-based analyzers (DEFAULT)
+├── analyzers.py             # Claude-based analyzers (optional)
+├── security.py              # Security validation
+├── github_client.py         # GitHub API operations
+├── report_generator.py      # Report formatting
+├── formatters.py            # Data formatting utilities
+├── claude_runner.py         # Claude CLI runner (optional)
+└── tests/                   # Test suite
+```
+
+### Switching to Claude-Based Analysis
+
+To use Claude AI for more sophisticated analysis:
+
+```bash
+export USE_CLAUDE_ANALYZERS=1
+```
+
+## Security Controls
+
+Implements comprehensive security controls per requirements:
+
+- **M1.2**: Input validation for PR numbers, data structures
+- **M2.1**: Sanitization of markdown content to prevent injection
+- **M3.1**: Read-only GitHub operations
+- **M3.2**: Label operations limited to allowed prefixes
+- **M3.4**: Clear validation rules with audit logging
+
+See `security.py` for full implementation.
+
+## Usage
+
+### Command Line
+
+```bash
+# Set PR number
+export PR_NUMBER=123
+
+# Run triage
+python3 .github/scripts/pr_triage_main.py
+```
+
+### GitHub Actions
+
+```yaml
+- name: PR Triage
+  env:
+    PR_NUMBER: ${{ github.event.pull_request.number }}
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  run: python3 .github/scripts/pr_triage_main.py
+```
+
+### Programmatic
+
+```python
+from pr_triage import PRTriageValidator
+
+validator = PRTriageValidator(pr_number=123)
+
+# Fetch PR data
+pr_data = validator.get_pr_data()
+
+# Run validations
+compliance = validator.validate_workflow_compliance(pr_data)
+labels = validator.detect_priority_complexity(pr_data)
+unrelated = validator.detect_unrelated_changes(pr_data)
+
+# Generate and post report
+report = validator.generate_triage_report(pr_data, compliance, labels, unrelated)
+validator.post_report(report)
+
+# Apply labels
+validator.apply_labels(labels)
+
+# Handle non-compliance
+if not compliance["overall_compliant"]:
+    validator.return_to_draft()
+```
+
+## MVP Heuristics
+
+### Workflow Compliance
+
+**Step 11 (Review)** - Requires review score >= 5 from:
+- Approved reviews (+5 each)
+- Review keywords: "code review", "security review", "lgtm", etc. (+1 each)
+
+**Step 12 (Feedback)** - Requires response score >= 3 from:
+- Response keywords: "addressed", "fixed", "updated", "implemented", etc.
+- Or 5+ total comments
+
+### Priority Detection
+
+- **CRITICAL**: Contains "security", "critical", "urgent", "vulnerability"
+- **HIGH**: Contains "bug", "fix", "error", "crash"
+- **LOW**: Contains "docs", "documentation", "typo", "cleanup"
+- **MEDIUM**: Default
+
+### Complexity Detection
+
+- **SIMPLE**: 1 file, < 50 lines
+- **MODERATE**: <= 3 files, < 200 lines
+- **COMPLEX**: <= 10 files, < 500 lines
+- **VERY_COMPLEX**: > 10 files or > 500 lines or architectural changes
+
+### Unrelated Changes
+
+Detected when PR has 3+ file categories:
+- Documentation
+- Tests
+- Configuration
+- Core code
+- Workflows
+
+## Testing
+
+```bash
+cd .github/scripts
+export PYTHONPATH=.
+
+# Run all tests
+python3 -m pytest pr_triage/tests/ -v
+
+# Run specific test module
+python3 -m pytest pr_triage/tests/test_security.py -v
+python3 -m pytest pr_triage/tests/test_analyzers_mvp.py -v
+```
+
+## Output
+
+### Triage Report
+
+Posted as PR comment with:
+- ✅/❌ Workflow compliance status
+- 🏷️ Priority and complexity labels
+- 🔍 Unrelated changes detection
+- 💡 Recommendations
+- 📊 Statistics
+
+### Audit Log
+
+Security audit trail in `.claude/runtime/logs/pr-triage-{number}/audit.log`:
+
+```
+[2025-11-23 23:45:12] PR-123 | get_pr_data | success | {'num_files': 5}
+[2025-11-23 23:45:15] PR-123 | apply_labels | success | {'labels': ['priority:high', 'complexity:moderate']}
+[2025-11-23 23:45:16] PR-123 | post_report | success
+```
+
+### Log File
+
+Detailed execution log in `.claude/runtime/logs/pr-triage-{number}/triage.log`:
+
+```
+[23:45:12] [INFO] [PR-123] Fetching PR data...
+[23:45:13] [INFO] [PR-123] Validating workflow compliance...
+[23:45:14] [INFO] [PR-123] Detecting priority and complexity...
+[23:45:15] [INFO] [PR-123] Applying labels...
+[23:45:16] [INFO] [PR-123] Posted triage report to PR #123
+[23:45:16] [INFO] [PR-123] PR triage completed successfully
+```
+
+## Error Handling
+
+- **Validation errors**: Logged with details, execution continues where safe
+- **GitHub API errors**: Logged as warnings, non-critical operations continue
+- **Security violations**: Immediately raise ValueError with audit trail
+- **All operations**: Wrapped in try-except with logging
+
+## Module Boundaries
+
+### Public Interface (`__init__.py`)
+
+```python
+from .validator import PRTriageValidator
+__all__ = ["PRTriageValidator"]
+```
+
+### Module Responsibilities
+
+- **validator.py**: Orchestration and workflow
+- **analyzers_mvp.py**: Heuristic analysis logic
+- **security.py**: All security validations
+- **github_client.py**: GitHub API operations via gh CLI
+- **report_generator.py**: Report formatting
+- **formatters.py**: Data formatting utilities
+
+## Dependencies
+
+- Python 3.11+
+- `gh` CLI (GitHub CLI) - must be authenticated
+- `pytest` (for tests)
+
+No external Python dependencies required for MVP mode.
+
+## Future Enhancements
+
+1. **Claude Integration**: More sophisticated analysis when enabled
+2. **Auto-fix**: Spawn Claude agent to fix compliance issues
+3. **Machine Learning**: Train model on historical PRs
+4. **Custom Rules**: User-defined validation rules
+5. **Metrics Dashboard**: Track PR health over time
+
+## Philosophy Alignment
+
+- **Zero-BS**: No stubs, all functions work or don't exist
+- **Modular**: Clear module boundaries and responsibilities
+- **Security First**: Comprehensive input validation and sanitization
+- **Fail-Safe**: Graceful degradation, never block PRs incorrectly
+- **Testable**: 27 tests covering security and analysis logic

--- a/.github/scripts/pr_triage/analyzers_mvp.py
+++ b/.github/scripts/pr_triage/analyzers_mvp.py
@@ -1,0 +1,327 @@
+"""PR analysis functions using heuristic-based validation (MVP).
+
+This MVP version uses simple heuristics instead of Claude AI for faster,
+deterministic validation. Suitable for initial deployment and testing.
+"""
+
+import re
+from typing import Any, Dict
+
+from .formatters import format_comments, format_files, format_reviews
+
+
+def validate_workflow_compliance(pr_data: Dict[str, Any]) -> Dict[str, Any]:
+    """Check if PR completed Steps 11-12 of workflow using heuristics.
+
+    Args:
+        pr_data: PR data dictionary
+
+    Returns:
+        Validation result with compliance status and details
+    """
+    comments = pr_data.get("comments", [])
+    reviews = pr_data.get("reviews", [])
+
+    # Step 11: Check for comprehensive code review
+    step11_completed = False
+    step11_evidence = ""
+
+    # Look for review indicators in comments and reviews
+    review_keywords = [
+        "code review",
+        "security review",
+        "philosophy compliance",
+        "test coverage",
+        "code quality",
+        "lgtm",
+        "looks good",
+        "approved",
+    ]
+
+    review_score = 0
+    for comment in comments:
+        body = comment.get("body", "").lower()
+        for keyword in review_keywords:
+            if keyword in body:
+                review_score += 1
+
+    for review in reviews:
+        body = review.get("body", "").lower()
+        state = review.get("state", "")
+
+        # Approved reviews count heavily
+        if state == "APPROVED":
+            review_score += 5
+
+        # Look for review keywords
+        for keyword in review_keywords:
+            if keyword in body:
+                review_score += 1
+
+    # Step 11 is complete if we have:
+    # - At least one approval OR
+    # - Multiple review-related comments
+    if review_score >= 5:
+        step11_completed = True
+        step11_evidence = (
+            f"Found {len(reviews)} reviews with review indicators. "
+            f"Review score: {review_score}"
+        )
+    else:
+        step11_evidence = (
+            f"Insufficient review evidence. Found {len(reviews)} reviews. "
+            f"Review score: {review_score} (need >= 5)"
+        )
+
+    # Step 12: Check for feedback implementation
+    step12_completed = False
+    step12_evidence = ""
+
+    # Look for response indicators
+    response_keywords = [
+        "addressed",
+        "fixed",
+        "updated",
+        "implemented",
+        "resolved",
+        "done",
+        "completed",
+    ]
+
+    response_score = 0
+    for comment in comments:
+        body = comment.get("body", "").lower()
+        for keyword in response_keywords:
+            if keyword in body:
+                response_score += 1
+
+    # Step 12 is complete if we have feedback responses
+    if response_score >= 3 or len(comments) > 5:
+        step12_completed = True
+        step12_evidence = (
+            f"Found {response_score} response indicators across {len(comments)} comments"
+        )
+    else:
+        step12_evidence = (
+            f"Insufficient feedback implementation. Response score: {response_score} (need >= 3)"
+        )
+
+    # Overall compliance
+    overall_compliant = step11_completed and step12_completed
+
+    blocking_issues = []
+    if not step11_completed:
+        blocking_issues.append(
+            "Step 11 incomplete: Need comprehensive code review with security, "
+            "quality, and philosophy checks"
+        )
+    if not step12_completed:
+        blocking_issues.append(
+            "Step 12 incomplete: Need to address and respond to review feedback"
+        )
+
+    recommendations = []
+    if not overall_compliant:
+        recommendations.append(
+            "Complete workflow steps 11-12 before marking PR as ready"
+        )
+    if len(reviews) == 0:
+        recommendations.append("Add at least one formal code review")
+
+    return {
+        "step11_completed": step11_completed,
+        "step11_evidence": step11_evidence,
+        "step12_completed": step12_completed,
+        "step12_evidence": step12_evidence,
+        "overall_compliant": overall_compliant,
+        "blocking_issues": blocking_issues,
+        "recommendations": recommendations,
+    }
+
+
+def detect_priority_complexity(pr_data: Dict[str, Any]) -> Dict[str, str]:
+    """Detect appropriate priority and complexity labels using heuristics.
+
+    Args:
+        pr_data: PR data dictionary
+
+    Returns:
+        Dictionary with priority and complexity labels
+    """
+    title = pr_data.get("title", "").lower()
+    body = pr_data.get("body", "").lower()
+    files = pr_data.get("files", [])
+    diff = pr_data.get("diff", "")
+
+    # Calculate metrics
+    num_files = len(files)
+    total_additions = sum(f.get("additions", 0) for f in files)
+    total_deletions = sum(f.get("deletions", 0) for f in files)
+    total_changes = total_additions + total_deletions
+
+    # Priority detection
+    priority = "MEDIUM"
+    priority_reasoning = "Default priority for normal changes"
+
+    critical_keywords = ["security", "critical", "urgent", "hotfix", "vulnerability", "data loss"]
+    high_keywords = ["bug", "fix", "error", "crash", "broken", "important"]
+    low_keywords = ["docs", "documentation", "readme", "typo", "comment", "cleanup"]
+
+    if any(kw in title or kw in body for kw in critical_keywords):
+        priority = "CRITICAL"
+        priority_reasoning = "Contains critical/security keywords"
+    elif any(kw in title or kw in body for kw in high_keywords):
+        priority = "HIGH"
+        priority_reasoning = "Bug fix or important change"
+    elif any(kw in title or kw in body for kw in low_keywords):
+        priority = "LOW"
+        priority_reasoning = "Documentation or minor cleanup"
+
+    # Complexity detection
+    complexity = "MODERATE"
+    complexity_reasoning = "Default complexity for normal changes"
+
+    if num_files == 1 and total_changes < 50:
+        complexity = "SIMPLE"
+        complexity_reasoning = f"Single file with {total_changes} lines changed"
+    elif num_files <= 3 and total_changes < 200:
+        complexity = "MODERATE"
+        complexity_reasoning = f"{num_files} files with {total_changes} lines changed"
+    elif num_files <= 10 and total_changes < 500:
+        complexity = "COMPLEX"
+        complexity_reasoning = f"{num_files} files with {total_changes} lines changed"
+    else:
+        complexity = "VERY_COMPLEX"
+        complexity_reasoning = (
+            f"{num_files} files with {total_changes} lines changed - "
+            f"system-wide changes"
+        )
+
+    # Check for architectural changes
+    architectural_files = [
+        f for f in files
+        if any(
+            pattern in f.get("path", "").lower()
+            for pattern in ["architecture", "design", "workflow", "config", "settings"]
+        )
+    ]
+    if architectural_files:
+        complexity = "VERY_COMPLEX"
+        complexity_reasoning += " (architectural changes detected)"
+
+    return {
+        "priority": priority,
+        "priority_reasoning": priority_reasoning,
+        "complexity": complexity,
+        "complexity_reasoning": complexity_reasoning,
+    }
+
+
+def detect_unrelated_changes(pr_data: Dict[str, Any]) -> Dict[str, Any]:
+    """Detect if PR contains unrelated changes using heuristics.
+
+    Args:
+        pr_data: PR data dictionary
+
+    Returns:
+        Dictionary with unrelated changes detection results
+    """
+    title = pr_data.get("title", "").lower()
+    body = pr_data.get("body", "").lower()
+    files = pr_data.get("files", [])
+
+    # Categorize files by type
+    file_categories = {
+        "docs": [],
+        "tests": [],
+        "config": [],
+        "core": [],
+        "workflows": [],
+        "other": [],
+    }
+
+    for file_data in files:
+        path = file_data.get("path", "").lower()
+
+        if "readme" in path or "doc" in path or path.endswith(".md"):
+            file_categories["docs"].append(path)
+        elif "test" in path or path.startswith("tests/"):
+            file_categories["tests"].append(path)
+        elif (
+            "config" in path
+            or path.endswith(".json")
+            or path.endswith(".yaml")
+            or path.endswith(".yml")
+            or path.endswith(".toml")
+        ):
+            file_categories["config"].append(path)
+        elif ".github/workflows" in path:
+            file_categories["workflows"].append(path)
+        elif path.endswith((".py", ".js", ".ts", ".go", ".java", ".cpp", ".c")):
+            file_categories["core"].append(path)
+        else:
+            file_categories["other"].append(path)
+
+    # Count non-empty categories
+    non_empty_categories = [cat for cat, paths in file_categories.items() if paths]
+
+    # Determine primary purpose from title
+    primary_purpose = "General code changes"
+    if "fix" in title or "bug" in title:
+        primary_purpose = "Bug fix"
+    elif "feat" in title or "add" in title:
+        primary_purpose = "New feature"
+    elif "refactor" in title:
+        primary_purpose = "Code refactoring"
+    elif "doc" in title:
+        primary_purpose = "Documentation update"
+    elif "test" in title:
+        primary_purpose = "Test updates"
+
+    # Detect unrelated changes
+    has_unrelated_changes = False
+    unrelated_files = []
+    unrelated_purposes = []
+    recommendation = "Changes appear focused"
+
+    # If we have more than 2 categories, likely unrelated
+    if len(non_empty_categories) >= 3:
+        has_unrelated_changes = True
+
+        # Identify what seems unrelated
+        if file_categories["docs"] and primary_purpose != "Documentation update":
+            unrelated_purposes.append("Documentation changes")
+            unrelated_files.extend(file_categories["docs"])
+
+        if file_categories["workflows"] and "workflow" not in title and "ci" not in title:
+            unrelated_purposes.append("Workflow/CI changes")
+            unrelated_files.extend(file_categories["workflows"])
+
+        if file_categories["config"] and "config" not in title:
+            unrelated_purposes.append("Configuration changes")
+            unrelated_files.extend(file_categories["config"])
+
+        recommendation = (
+            "Consider splitting this PR into separate focused PRs for each concern"
+        )
+
+    # Special case: Large refactoring with feature work
+    if (
+        "refactor" in title
+        and ("feat" in title or "add" in title)
+        and len(files) > 5
+    ):
+        has_unrelated_changes = True
+        unrelated_purposes.append("Mixed refactoring and feature work")
+        recommendation = "Separate refactoring from new features into different PRs"
+
+    return {
+        "has_unrelated_changes": has_unrelated_changes,
+        "unrelated_files": unrelated_files[:10],  # Limit to 10 files
+        "primary_purpose": primary_purpose,
+        "unrelated_purposes": unrelated_purposes,
+        "recommendation": recommendation,
+        "file_categories": {
+            k: len(v) for k, v in file_categories.items() if v
+        },  # Statistics
+    }

--- a/.github/scripts/pr_triage/security.py
+++ b/.github/scripts/pr_triage/security.py
@@ -1,0 +1,253 @@
+"""Security validation for PR triage automation.
+
+Implements defense controls per security requirements:
+- M1.2: Input validation for PR data
+- M2.1: Sanitization of untrusted content
+- M3.1: Read-only GitHub operations
+- M3.2: Label operations only
+- M3.4: Clear validation rules
+"""
+
+import os
+import re
+from typing import Any, Dict, List
+
+
+def validate_pr_number(pr_number: int) -> None:
+    """Validate PR number is within acceptable range.
+
+    Args:
+        pr_number: PR number to validate
+
+    Raises:
+        ValueError: If PR number is invalid
+    """
+    if not isinstance(pr_number, int):
+        raise ValueError(f"PR number must be integer, got {type(pr_number)}")
+
+    if pr_number <= 0:
+        raise ValueError(f"PR number must be positive, got {pr_number}")
+
+    if pr_number > 999999:
+        raise ValueError(f"PR number too large: {pr_number}")
+
+
+def validate_github_token() -> str:
+    """Validate GitHub token exists and is minimally formatted.
+
+    Returns:
+        GitHub token from environment
+
+    Raises:
+        ValueError: If token is missing or invalid
+    """
+    token = os.environ.get("GITHUB_TOKEN", "")
+
+    if not token:
+        raise ValueError("GITHUB_TOKEN environment variable not set")
+
+    # Basic format check (gh_ or gho_ prefix)
+    if not (token.startswith("gh_") or token.startswith("gho_")):
+        raise ValueError("GITHUB_TOKEN appears malformed (wrong prefix)")
+
+    if len(token) < 20:
+        raise ValueError("GITHUB_TOKEN appears too short")
+
+    return token
+
+
+def sanitize_markdown(text: str) -> str:
+    """Sanitize markdown text to prevent injection attacks.
+
+    Args:
+        text: Raw markdown text
+
+    Returns:
+        Sanitized markdown text
+    """
+    if not isinstance(text, str):
+        return ""
+
+    # Remove HTML tags (except safe ones)
+    safe_tags = ["b", "i", "em", "strong", "code", "pre", "a", "ul", "ol", "li"]
+    safe_pattern = "|".join(safe_tags)
+
+    # Remove all HTML tags except safe ones
+    text = re.sub(
+        r"<(?!/?({})\b)[^>]*>".format(safe_pattern),
+        "",
+        text,
+        flags=re.IGNORECASE,
+    )
+
+    # Remove script tags completely
+    text = re.sub(r"<script[^>]*>.*?</script>", "", text, flags=re.IGNORECASE | re.DOTALL)
+
+    # Remove event handlers
+    text = re.sub(r"\bon\w+\s*=", "", text, flags=re.IGNORECASE)
+
+    # Limit length to prevent DOS
+    max_length = 100000
+    if len(text) > max_length:
+        text = text[:max_length] + "\n\n[Content truncated for safety]"
+
+    return text
+
+
+def validate_label_name(label: str) -> None:
+    """Validate label name is safe to use.
+
+    Args:
+        label: Label name to validate
+
+    Raises:
+        ValueError: If label name is invalid
+    """
+    if not isinstance(label, str):
+        raise ValueError(f"Label must be string, got {type(label)}")
+
+    if not label:
+        raise ValueError("Label cannot be empty")
+
+    if len(label) > 100:
+        raise ValueError(f"Label too long: {len(label)} chars")
+
+    # Only allow alphanumeric, dash, underscore, colon
+    if not re.match(r"^[a-zA-Z0-9_:-]+$", label):
+        raise ValueError(f"Label contains invalid characters: {label}")
+
+
+def validate_allowed_labels(labels: List[str]) -> None:
+    """Validate labels are from allowed set.
+
+    Args:
+        labels: List of label names to validate
+
+    Raises:
+        ValueError: If any label is not allowed
+    """
+    allowed_prefixes = [
+        "priority:",
+        "complexity:",
+        "status:",
+        "type:",
+    ]
+
+    for label in labels:
+        validate_label_name(label)
+
+        # Check if label starts with allowed prefix
+        if not any(label.startswith(prefix) for prefix in allowed_prefixes):
+            raise ValueError(
+                f"Label '{label}' not allowed. Must start with: {allowed_prefixes}"
+            )
+
+
+def validate_pr_data(pr_data: Dict[str, Any]) -> None:
+    """Validate PR data structure is safe to process.
+
+    Args:
+        pr_data: PR data dictionary from GitHub
+
+    Raises:
+        ValueError: If PR data is invalid or unsafe
+    """
+    if not isinstance(pr_data, dict):
+        raise ValueError(f"PR data must be dict, got {type(pr_data)}")
+
+    required_fields = ["title", "body", "author", "files"]
+    for field in required_fields:
+        if field not in pr_data:
+            raise ValueError(f"PR data missing required field: {field}")
+
+    # Validate author structure
+    author = pr_data.get("author", {})
+    if not isinstance(author, dict) or "login" not in author:
+        raise ValueError("PR author data malformed")
+
+    # Validate files is list
+    files = pr_data.get("files", [])
+    if not isinstance(files, list):
+        raise ValueError("PR files must be list")
+
+    # Validate comments is list
+    comments = pr_data.get("comments", [])
+    if not isinstance(comments, list):
+        raise ValueError("PR comments must be list")
+
+    # Validate reviews is list
+    reviews = pr_data.get("reviews", [])
+    if not isinstance(reviews, list):
+        raise ValueError("PR reviews must be list")
+
+
+def is_safe_operation(operation: str) -> bool:
+    """Check if GitHub operation is allowed (read-only or label operations).
+
+    Args:
+        operation: Operation name to check
+
+    Returns:
+        True if operation is safe, False otherwise
+    """
+    safe_operations = [
+        "get_pr_data",
+        "apply_labels",
+        "post_comment",
+        "return_to_draft",
+    ]
+
+    return operation in safe_operations
+
+
+def validate_file_paths(files: List[Dict[str, Any]]) -> None:
+    """Validate file paths don't contain path traversal attacks.
+
+    Args:
+        files: List of file dictionaries with 'path' field
+
+    Raises:
+        ValueError: If any file path is unsafe
+    """
+    for file_data in files:
+        path = file_data.get("path", "")
+
+        if not isinstance(path, str):
+            raise ValueError(f"File path must be string, got {type(path)}")
+
+        # Check for path traversal
+        if ".." in path:
+            raise ValueError(f"Path traversal detected in file path: {path}")
+
+        # Check for absolute paths
+        if path.startswith("/"):
+            raise ValueError(f"Absolute path not allowed: {path}")
+
+        # Check for unusual characters
+        if re.search(r"[<>|;`$&]", path):
+            raise ValueError(f"Invalid characters in file path: {path}")
+
+
+def create_audit_log(
+    pr_number: int, operation: str, result: str, details: Dict[str, Any] = None
+) -> str:
+    """Create audit log entry for security tracking.
+
+    Args:
+        pr_number: PR number
+        operation: Operation performed
+        result: Result (success/failure)
+        details: Additional details
+
+    Returns:
+        Formatted audit log entry
+    """
+    import time
+
+    timestamp = time.strftime("%Y-%m-%d %H:%M:%S")
+    log_entry = f"[{timestamp}] PR-{pr_number} | {operation} | {result}"
+
+    if details:
+        log_entry += f" | {details}"
+
+    return log_entry

--- a/.github/scripts/pr_triage/tests/test_analyzers.py
+++ b/.github/scripts/pr_triage/tests/test_analyzers.py
@@ -107,13 +107,5 @@ def test_detect_unrelated_changes_structure():
 
 
 if __name__ == "__main__":
-    # Run tests
-    test_extract_json_from_code_block()
-    test_extract_json_from_raw()
-    test_extract_json_nested()
-    test_extract_json_no_match()
-    test_extract_json_malformed()
-    test_validate_workflow_compliance_structure()
-    test_detect_priority_complexity_structure()
-    test_detect_unrelated_changes_structure()
-    print("All analyzer tests passed!")
+    import pytest
+    pytest.main([__file__, "-v"])

--- a/.github/scripts/pr_triage/tests/test_analyzers_mvp.py
+++ b/.github/scripts/pr_triage/tests/test_analyzers_mvp.py
@@ -1,0 +1,215 @@
+"""Tests for MVP heuristic-based analyzers."""
+
+import pytest
+
+from pr_triage import analyzers_mvp
+
+
+def test_validate_workflow_compliance_with_approval():
+    """Test workflow compliance with approved review."""
+    pr_data = {
+        "title": "Test PR",
+        "body": "Test body",
+        "comments": [
+            {"body": "Code review completed. LGTM!"},
+            {"body": "Security review passed."},
+            {"body": "Addressed feedback"},
+            {"body": "Fixed the issue"},
+            {"body": "Updated per comments"},
+            {"body": "Done"},
+        ],
+        "reviews": [
+            {"state": "APPROVED", "body": "Looks good to me!"},
+        ],
+    }
+
+    result = analyzers_mvp.validate_workflow_compliance(pr_data)
+
+    assert result["step11_completed"] is True
+    assert result["overall_compliant"] is True
+    assert len(result["blocking_issues"]) == 0
+
+
+def test_validate_workflow_compliance_no_review():
+    """Test workflow compliance with no review."""
+    pr_data = {
+        "title": "Test PR",
+        "body": "Test body",
+        "comments": [],
+        "reviews": [],
+    }
+
+    result = analyzers_mvp.validate_workflow_compliance(pr_data)
+
+    assert result["step11_completed"] is False
+    assert result["overall_compliant"] is False
+    assert len(result["blocking_issues"]) > 0
+
+
+def test_validate_workflow_compliance_partial():
+    """Test workflow compliance with partial evidence."""
+    pr_data = {
+        "title": "Test PR",
+        "body": "Test body",
+        "comments": [
+            {"body": "Code quality looks good"},
+            {"body": "Test coverage verified"},
+        ],
+        "reviews": [],
+    }
+
+    result = analyzers_mvp.validate_workflow_compliance(pr_data)
+
+    # Should have some review score but not enough for compliance
+    assert result["overall_compliant"] is False
+    assert "score" in result["step11_evidence"].lower()
+
+
+def test_detect_priority_complexity_critical():
+    """Test priority detection for critical issues."""
+    pr_data = {
+        "title": "CRITICAL: Security vulnerability in auth",
+        "body": "Fixes critical security issue",
+        "files": [{"path": "auth.py", "additions": 10, "deletions": 5}],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_priority_complexity(pr_data)
+
+    assert result["priority"] == "CRITICAL"
+    assert "security" in result["priority_reasoning"].lower()
+
+
+def test_detect_priority_complexity_simple():
+    """Test complexity detection for simple changes."""
+    pr_data = {
+        "title": "Update typo in documentation",
+        "body": "Simple typo correction in docs",
+        "files": [{"path": "docs/README.md", "additions": 1, "deletions": 1}],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_priority_complexity(pr_data)
+
+    assert result["priority"] == "LOW"
+    assert result["complexity"] == "SIMPLE"
+
+
+def test_detect_priority_complexity_very_complex():
+    """Test complexity detection for large changes."""
+    pr_data = {
+        "title": "Major refactoring of core system",
+        "body": "Large architectural changes",
+        "files": [
+            {"path": f"file{i}.py", "additions": 50, "deletions": 30}
+            for i in range(15)
+        ],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_priority_complexity(pr_data)
+
+    assert result["complexity"] == "VERY_COMPLEX"
+    assert "15 files" in result["complexity_reasoning"]
+
+
+def test_detect_unrelated_changes_focused():
+    """Test detection with focused changes."""
+    pr_data = {
+        "title": "Add user authentication",
+        "body": "Implements JWT auth",
+        "files": [
+            {"path": "auth.py"},
+            {"path": "models/user.py"},
+            {"path": "tests/test_auth.py"},
+        ],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_unrelated_changes(pr_data)
+
+    assert result["has_unrelated_changes"] is False
+
+
+def test_detect_unrelated_changes_mixed():
+    """Test detection with mixed unrelated changes."""
+    pr_data = {
+        "title": "Add feature X",
+        "body": "New feature implementation",
+        "files": [
+            {"path": "feature_x.py"},
+            {"path": "tests/test_feature_x.py"},
+            {"path": "README.md"},
+            {"path": ".github/workflows/ci.yml"},
+            {"path": "config.json"},
+        ],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_unrelated_changes(pr_data)
+
+    assert result["has_unrelated_changes"] is True
+    assert len(result["unrelated_purposes"]) > 0
+
+
+def test_detect_unrelated_changes_refactor_and_feature():
+    """Test detection with mixed refactor and feature."""
+    pr_data = {
+        "title": "Refactor and add feature Y",
+        "body": "Refactoring + new feature",
+        "files": [{"path": f"file{i}.py"} for i in range(10)],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_unrelated_changes(pr_data)
+
+    assert result["has_unrelated_changes"] is True
+    assert "refactoring" in result["recommendation"].lower()
+
+
+def test_detect_priority_high_for_bugs():
+    """Test priority is HIGH for bug fixes."""
+    pr_data = {
+        "title": "Fix critical bug in payment processing",
+        "body": "Fixes issue #123",
+        "files": [{"path": "payment.py", "additions": 20, "deletions": 10}],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_priority_complexity(pr_data)
+
+    assert result["priority"] in ["HIGH", "CRITICAL"]
+
+
+def test_detect_complexity_moderate():
+    """Test complexity is MODERATE for typical changes."""
+    pr_data = {
+        "title": "Add validation function",
+        "body": "New validation logic",
+        "files": [
+            {"path": "validators.py", "additions": 80, "deletions": 20},
+            {"path": "tests/test_validators.py", "additions": 50, "deletions": 0},
+        ],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_priority_complexity(pr_data)
+
+    assert result["complexity"] == "MODERATE"
+
+
+def test_architectural_changes_increase_complexity():
+    """Test architectural files increase complexity."""
+    pr_data = {
+        "title": "Update workflow",
+        "body": "Workflow changes",
+        "files": [
+            {"path": "workflow/DEFAULT_WORKFLOW.md", "additions": 30, "deletions": 10},
+        ],
+        "diff": "test diff",
+    }
+
+    result = analyzers_mvp.detect_priority_complexity(pr_data)
+
+    assert result["complexity"] == "VERY_COMPLEX"
+    assert "architectural" in result["complexity_reasoning"].lower()

--- a/.github/scripts/pr_triage/tests/test_formatters.py
+++ b/.github/scripts/pr_triage/tests/test_formatters.py
@@ -123,17 +123,5 @@ def test_format_files_limits_count():
 
 
 if __name__ == "__main__":
-    # Run tests
-    test_format_comments_empty()
-    test_format_comments_single()
-    test_format_comments_multiple()
-    test_format_comments_truncates_long()
-    test_format_comments_limits_count()
-    test_format_reviews_empty()
-    test_format_reviews_single()
-    test_format_reviews_multiple_states()
-    test_format_files_empty()
-    test_format_files_single()
-    test_format_files_multiple()
-    test_format_files_limits_count()
-    print("All formatter tests passed!")
+    import pytest
+    pytest.main([__file__, "-v"])

--- a/.github/scripts/pr_triage/tests/test_report_generator.py
+++ b/.github/scripts/pr_triage/tests/test_report_generator.py
@@ -146,9 +146,5 @@ def test_generate_report_includes_statistics():
 
 
 if __name__ == "__main__":
-    # Run tests
-    test_generate_report_compliant()
-    test_generate_report_non_compliant()
-    test_generate_report_unrelated_changes()
-    test_generate_report_includes_statistics()
-    print("All report generator tests passed!")
+    import pytest
+    pytest.main([__file__, "-v"])

--- a/.github/scripts/pr_triage/tests/test_security.py
+++ b/.github/scripts/pr_triage/tests/test_security.py
@@ -1,0 +1,163 @@
+"""Tests for security validation module."""
+
+import os
+import pytest
+
+from pr_triage import security
+
+
+def test_validate_pr_number_valid():
+    """Test valid PR numbers pass."""
+    security.validate_pr_number(1)
+    security.validate_pr_number(123)
+    security.validate_pr_number(999999)
+
+
+def test_validate_pr_number_invalid():
+    """Test invalid PR numbers raise ValueError."""
+    with pytest.raises(ValueError, match="must be integer"):
+        security.validate_pr_number("123")
+
+    with pytest.raises(ValueError, match="must be positive"):
+        security.validate_pr_number(0)
+
+    with pytest.raises(ValueError, match="must be positive"):
+        security.validate_pr_number(-1)
+
+    with pytest.raises(ValueError, match="too large"):
+        security.validate_pr_number(1000000)
+
+
+def test_sanitize_markdown_removes_scripts():
+    """Test script tags are removed."""
+    text = "Hello <script>alert('xss')</script> world"
+    result = security.sanitize_markdown(text)
+    assert "<script>" not in result
+    assert "</script>" not in result
+
+
+def test_sanitize_markdown_removes_event_handlers():
+    """Test event handlers are removed."""
+    text = '<div onclick="alert(1)">Click me</div>'
+    result = security.sanitize_markdown(text)
+    assert "onclick" not in result
+
+
+def test_sanitize_markdown_limits_length():
+    """Test long text is truncated."""
+    long_text = "x" * 200000
+    result = security.sanitize_markdown(long_text)
+    assert len(result) <= 110000  # 100k + truncation message
+
+
+def test_validate_label_name_valid():
+    """Test valid label names pass."""
+    security.validate_label_name("priority:high")
+    security.validate_label_name("complexity:moderate")
+    security.validate_label_name("status:in-progress")
+
+
+def test_validate_label_name_invalid():
+    """Test invalid label names raise ValueError."""
+    with pytest.raises(ValueError, match="must be string"):
+        security.validate_label_name(123)
+
+    with pytest.raises(ValueError, match="cannot be empty"):
+        security.validate_label_name("")
+
+    with pytest.raises(ValueError, match="too long"):
+        security.validate_label_name("x" * 101)
+
+    with pytest.raises(ValueError, match="invalid characters"):
+        security.validate_label_name("label with spaces")
+
+    with pytest.raises(ValueError, match="invalid characters"):
+        security.validate_label_name("label;drop table")
+
+
+def test_validate_allowed_labels_valid():
+    """Test allowed labels pass."""
+    labels = ["priority:high", "complexity:moderate"]
+    security.validate_allowed_labels(labels)
+
+
+def test_validate_allowed_labels_invalid():
+    """Test disallowed labels raise ValueError."""
+    with pytest.raises(ValueError, match="not allowed"):
+        security.validate_allowed_labels(["random:label"])
+
+    with pytest.raises(ValueError, match="not allowed"):
+        security.validate_allowed_labels(["malicious-label"])
+
+
+def test_validate_pr_data_valid():
+    """Test valid PR data passes."""
+    pr_data = {
+        "title": "Test PR",
+        "body": "Description",
+        "author": {"login": "testuser"},
+        "files": [],
+        "comments": [],
+        "reviews": [],
+    }
+    security.validate_pr_data(pr_data)
+
+
+def test_validate_pr_data_invalid():
+    """Test invalid PR data raises ValueError."""
+    with pytest.raises(ValueError, match="must be dict"):
+        security.validate_pr_data("not a dict")
+
+    with pytest.raises(ValueError, match="missing required field"):
+        security.validate_pr_data({})
+
+    with pytest.raises(ValueError, match="author data malformed"):
+        security.validate_pr_data({
+            "title": "Test",
+            "body": "Body",
+            "author": "notadict",
+            "files": [],
+        })
+
+
+def test_validate_file_paths_valid():
+    """Test valid file paths pass."""
+    files = [
+        {"path": "src/main.py"},
+        {"path": "tests/test_main.py"},
+        {"path": ".github/workflows/ci.yml"},
+    ]
+    security.validate_file_paths(files)
+
+
+def test_validate_file_paths_invalid():
+    """Test invalid file paths raise ValueError."""
+    with pytest.raises(ValueError, match="Path traversal"):
+        security.validate_file_paths([{"path": "../etc/passwd"}])
+
+    with pytest.raises(ValueError, match="Absolute path"):
+        security.validate_file_paths([{"path": "/etc/passwd"}])
+
+    with pytest.raises(ValueError, match="Invalid characters"):
+        security.validate_file_paths([{"path": "file;rm -rf /"}])
+
+
+def test_is_safe_operation():
+    """Test safe operation checking."""
+    assert security.is_safe_operation("get_pr_data")
+    assert security.is_safe_operation("apply_labels")
+    assert security.is_safe_operation("post_comment")
+    assert security.is_safe_operation("return_to_draft")
+
+    assert not security.is_safe_operation("delete_repo")
+    assert not security.is_safe_operation("force_merge")
+
+
+def test_create_audit_log():
+    """Test audit log creation."""
+    log = security.create_audit_log(123, "test_op", "success", {"key": "value"})
+
+    assert "PR-123" in log
+    assert "test_op" in log
+    assert "success" in log
+    assert "{'key': 'value'}" in log

--- a/.github/scripts/pr_triage/validator.py
+++ b/.github/scripts/pr_triage/validator.py
@@ -1,28 +1,46 @@
 """PR triage validator - main orchestration."""
 
+import os
 import time
 from pathlib import Path
 from typing import Any, Dict
 
-from . import analyzers, github_client
-from .claude_runner import run_claude
+from . import github_client, security
 from .report_generator import generate_triage_report
+
+# Use MVP analyzers (heuristic-based) by default
+# Can switch to Claude-based analyzers by setting USE_CLAUDE_ANALYZERS=1
+USE_CLAUDE = os.environ.get("USE_CLAUDE_ANALYZERS", "0") == "1"
+
+if USE_CLAUDE:
+    from . import analyzers
+else:
+    from . import analyzers_mvp as analyzers
 
 REPO_ROOT = Path(__file__).parent.parent.parent.parent
 
 
 class PRTriageValidator:
-    """Validates PR compliance using Claude CLI."""
+    """Validates PR compliance using heuristic analysis or Claude CLI."""
 
     def __init__(self, pr_number: int):
         """Initialize validator.
 
         Args:
             pr_number: GitHub PR number
+
+        Raises:
+            ValueError: If PR number is invalid
         """
+        # Security validation
+        security.validate_pr_number(pr_number)
+
         self.pr_number = pr_number
         self.log_dir = REPO_ROOT / ".claude" / "runtime" / "logs" / f"pr-triage-{pr_number}"
         self.log_dir.mkdir(parents=True, exist_ok=True)
+
+        # Audit log file
+        self.audit_log = self.log_dir / "audit.log"
 
     def log(self, msg: str, level: str = "INFO") -> None:
         """Log message to console and file.
@@ -31,6 +49,9 @@ class PRTriageValidator:
             msg: Message to log
             level: Log level
         """
+        # Sanitize log message
+        msg = security.sanitize_markdown(msg)
+
         log_msg = f"[{time.strftime('%H:%M:%S')}] [{level}] [PR-{self.pr_number}] {msg}"
         print(log_msg)
 
@@ -38,9 +59,46 @@ class PRTriageValidator:
         with open(log_file, "a") as f:
             f.write(log_msg + "\n")
 
+    def audit(self, operation: str, result: str, details: Dict[str, Any] = None) -> None:
+        """Write audit log entry.
+
+        Args:
+            operation: Operation name
+            result: Result status
+            details: Additional details
+        """
+        entry = security.create_audit_log(self.pr_number, operation, result, details)
+        with open(self.audit_log, "a") as f:
+            f.write(entry + "\n")
+
     def get_pr_data(self) -> Dict[str, Any]:
-        """Fetch PR data using gh CLI."""
-        return github_client.get_pr_data(self.pr_number)
+        """Fetch PR data using gh CLI.
+
+        Returns:
+            PR data dictionary
+
+        Raises:
+            ValueError: If PR data is invalid
+        """
+        self.audit("get_pr_data", "started")
+
+        try:
+            pr_data = github_client.get_pr_data(self.pr_number)
+
+            # Security validation
+            security.validate_pr_data(pr_data)
+            security.validate_file_paths(pr_data.get("files", []))
+
+            # Sanitize text fields
+            pr_data["title"] = security.sanitize_markdown(pr_data.get("title", ""))
+            pr_data["body"] = security.sanitize_markdown(pr_data.get("body", ""))
+
+            self.audit("get_pr_data", "success", {"num_files": len(pr_data.get("files", []))})
+            return pr_data
+
+        except Exception as e:
+            self.audit("get_pr_data", "failed", {"error": str(e)})
+            raise
 
     def validate_workflow_compliance(self, pr_data: Dict[str, Any]) -> Dict[str, Any]:
         """Check if PR completed Steps 11-12 of workflow."""
@@ -65,7 +123,14 @@ class PRTriageValidator:
         return generate_triage_report(self.pr_number, pr_data, compliance, labels, unrelated)
 
     def apply_labels(self, labels: Dict[str, str]) -> None:
-        """Apply priority and complexity labels to PR."""
+        """Apply priority and complexity labels to PR.
+
+        Args:
+            labels: Dictionary with priority and complexity labels
+
+        Raises:
+            ValueError: If labels are invalid
+        """
         priority = labels.get("priority", "MEDIUM")
         complexity = labels.get("complexity", "MODERATE")
 
@@ -74,63 +139,48 @@ class PRTriageValidator:
             f"complexity:{complexity.lower()}",
         ]
 
+        # Security validation
+        security.validate_allowed_labels(label_names)
+
+        self.audit("apply_labels", "started", {"labels": label_names})
+
         try:
             github_client.apply_labels(self.pr_number, label_names)
+            self.audit("apply_labels", "success", {"labels": label_names})
         except Exception as e:
+            self.audit("apply_labels", "failed", {"error": str(e)})
             self.log(f"Warning: Failed to apply labels: {e}", level="WARNING")
 
     def return_to_draft(self) -> None:
         """Convert PR back to draft status."""
+        self.audit("return_to_draft", "started")
+
         try:
             github_client.return_to_draft(self.pr_number)
             self.log(f"Returned PR #{self.pr_number} to draft status")
+            self.audit("return_to_draft", "success")
         except Exception as e:
+            self.audit("return_to_draft", "failed", {"error": str(e)})
             self.log(f"Warning: Failed to return to draft: {e}", level="WARNING")
 
-    def spawn_auto_fix(self, compliance: Dict[str, Any]) -> None:
-        """Spawn auto mode to fix workflow compliance issues."""
-        issues = compliance.get("blocking_issues", [])
-        if not issues:
-            return
-
-        fix_prompt = f"""Fix workflow compliance issues in PR #{self.pr_number}.
-
-**Blocking Issues:**
-{chr(10).join(f"- {issue}" for issue in issues)}
-
-**Required Actions:**
-1. Complete Step 11 (Review the PR):
-   - Post comprehensive code review
-   - Perform security review
-   - Verify code quality and standards
-   - Check philosophy compliance
-   - Verify test coverage
-
-2. Complete Step 12 (Implement Review Feedback):
-   - Address all review comments
-   - Push updates to PR
-   - Respond to comments
-   - Ensure tests still pass
-
-Follow the DEFAULT_WORKFLOW.md process exactly.
-"""
-
-        self.log("Spawning auto mode to fix compliance issues...")
-
-        result = run_claude(fix_prompt, timeout=1800)
-
-        if result["exit_code"] == 0:
-            self.log("Auto-fix completed successfully")
-        else:
-            self.log(f"Auto-fix failed: {result['stderr']}", level="ERROR")
-
     def post_report(self, report: str) -> None:
-        """Post triage report as PR comment."""
+        """Post triage report as PR comment.
+
+        Args:
+            report: Markdown report to post
+        """
+        # Sanitize report content
+        report = security.sanitize_markdown(report)
+
         report_file = self.log_dir / "triage_report.md"
         report_file.write_text(report)
+
+        self.audit("post_report", "started", {"report_length": len(report)})
 
         try:
             github_client.post_comment(self.pr_number, str(report_file))
             self.log(f"Posted triage report to PR #{self.pr_number}")
+            self.audit("post_report", "success")
         except Exception as e:
+            self.audit("post_report", "failed", {"error": str(e)})
             self.log(f"Error posting report: {e}", level="ERROR")

--- a/.github/scripts/pr_triage_main.py
+++ b/.github/scripts/pr_triage_main.py
@@ -59,11 +59,14 @@ def main():
         # Handle non-compliance
         if not compliance.get("overall_compliant"):
             validator.log(
-                "PR is non-compliant, returning to draft and spawning auto-fix...",
+                "PR is non-compliant, returning to draft status...",
                 level="WARNING",
             )
             validator.return_to_draft()
-            validator.spawn_auto_fix(compliance)
+            validator.log(
+                "NOTE: Auto-fix requires Claude CLI. Set USE_CLAUDE_ANALYZERS=1 to enable.",
+                level="INFO",
+            )
 
         validator.log("PR triage completed successfully")
 

